src/foundry/client-esm/applications/dice/roll-resolver.d.mts

```diff
@@ -1,4 +1,107 @@
+import type { DeepPartial } from "../../../../types/utils.d.mts";
+import type { ApplicationConfiguration } from "../../../common/config.d.mts";
 import type ApplicationV2 from "../api/application.d.mts";
 import type HandlebarsApplicationMixin from "../api/handlebars-application.mjs";
 
-export default class RollResolver extends HandlebarsApplicationMixin(ApplicationV2) {}
+/**
+ * An application responsible for handling unfulfilled dice terms in a roll.
+ */
+
+declare class RollResolver<Configuration extends ApplicationConfiguration> extends HandlebarsApplicationMixin(
+  ApplicationV2,
+) {
+  constructor(roll: Roll, options?: DeepPartial<Configuration>);
+
+  /**
+   * A collection of fulfillable dice terms.
+   */
+  get fulfillable(): Map<string, RollResolver.DiceTermFulfillmentDescriptor>;
+  #fulfillable: Map<string, RollResolver.DiceTermFulfillmentDescriptor>;
+
+  /**
+   * A function to call when the first pass of fulfillment is complete.
+   */
+  #resolve?: () => Promise<void>;
+
+  /**
+   * The roll being resolved.
+   */
+  get roll(): Roll;
+  #roll: Roll;
+
+  /**
+   * Identify any terms in this Roll that should be fulfilled externally, and prompt the user to do so.
+   * @returns Returns a Promise that resolves when the first pass of fulfillment is complete.
+   */
+  awaitFulfillment(): Promise<void>;
+
+  /**
+   * Register a fulfilled die roll.
+   * @param method        - The method used for fulfillment.
+   * @param denomination  - The denomination of the fulfilled die.
+   * @param result        - The rolled number.
+   * @returns               Whether the result was consumed.
+   */
+  registerResult(method: string, denomination: string, result: number): boolean;
+
+  /**
+   * Handle prompting for a single extra result from a term.
+   * @param term                - The term.
+   * @param method              - The method used to obtain the result.
+   * @param  options            - Optional configuration options
+   * @param  options.reroll     - Defaults to false
+   * @param  options.explode    - Defaults to false
+   * @returns
+   */
+  resolveResult(
+    term: DiceTerm,
+    method: string,
+    options?: { reroll?: boolean; explode?: boolean },
```

Comment:
So TypeScript has something called `exactOptionalPropertyTypes` that some of our users use.

This means that for properties in _objects_ i.e. not in function parameters, there's a difference between `{ reroll?: boolean; explode?: boolean }` and `{ reroll?: boolean | undefined; explode?: boolean | undefined }`

There's _not_ a difference between `options?: T` and `options?: T | undefined` though. The reason why is because of the Javascript `in` operation that checks if a key exists in the object, thusly making it possible to distinguish between `{ reroll: undefined }` and `{}` but `in` can't be applied to functions or elsewhere.

This is such a common thing that we already have a helper type; `InexactPartial<{ reroll: boolean, explode: boolean }>` means you won't have to explicitly write out the `| undefined`.